src/annotator/util/test/scroll-test.js

```diff
@@ -68,4 +72,59 @@ describe('annotator/util/scroll', () => {
       container.remove();
     });
   });
+
+  describe('scrollElementIntoView', () => {
+    let container;
+    let target;
+
+    beforeEach(() => {
+      container = document.createElement('div');
+      container.style.height = '500px';
+      container.style.overflow = 'auto';
+      container.style.position = 'relative';
+      document.body.append(container);
+
+      target = document.createElement('div');
+      target.style.position = 'absolute';
+      target.style.top = '1000px';
+      target.style.height = '20px';
+      target.style.width = '100px';
+      container.append(target);
+
+      assert.isTrue(container.scrollHeight > container.clientHeight);
+    });
+
+    afterEach(() => {
+      target.remove();
+    });
+
+    // A basic test for scrolling. We assume that the underlying implementation
+    // has more detailed tests.
+    it('scrolls element into view', async () => {
+      await scrollElementIntoView(target, { maxDuration: 1 });
+
+      const containerRect = container.getBoundingClientRect();
+      const targetRect = target.getBoundingClientRect();
+
+      assert.isTrue(containerRect.top <= targetRect.top);
+      assert.isTrue(containerRect.bottom >= targetRect.bottom);
+    });
+
+    it('installs scroll-into-view workaround for XHTML documents', async () => {
+      try {
+        Object.defineProperty(document.body, 'tagName', {
+          value: 'body',
```

Comment:
Maybe drop a comment somewhere near here that the `body` tagname is lowercase in XHTML documents as a reminder?